test/models/concerns/smart_proxy_extensions_test.rb

```diff
@@ -60,4 +60,22 @@ def test_save_with_organization_location
       assert_equal @proxy_mirror.lifecycle_environments.all.count, 0
     end
   end
+
+  class SmartProxyPulp3Test < ActiveSupport::TestCase
+    def setup
+      @master = FactoryBot.create(:smart_proxy, :default_smart_proxy, :with_pulp3)
```

Comment:
It's probably the right time to create a fixture for a pulp3 proxy (maybe there already is one!)

Reply:
Its tricky because you can only have 1 'default' smart proxy or things get really wonky, so if we have a default one in fixtures then we'd have to delete it or modify it anytime we wanted to test pulp2.  

We'd also have to figure out to add onto the existing foreman fixtures since they are defined there (and fixtures don't have a way to do that easily like factories).

Reply:
Oh good points. Probably more trouble than it's worth right now :)